node-js-basics-101/node_modules/uws/uws.js

```diff
@@ -0,0 +1,563 @@
+            native.client.terminate(this.external);
+            this.external = null;
+        }
+    }
+
+    send(message, options, cb) {
+        if (this.external) {
+            if (typeof options === 'function') {
+                cb = options;
+                options = null;
+            }
+
+            const binary = options && options.binary || typeof message !== 'string';
+
+            native.client.send(this.external, message, binary ? WebSocketClient.OPCODE_BINARY : WebSocketClient.OPCODE_TEXT, cb ? (() => {
+                process.nextTick(cb);
+            }) : undefined);
+        } else if (cb) {
+            cb(new Error('not opened'));
+        }
+    }
+
+    close(code, data) {
+        if (this.external) {
+            native.client.close(this.external, code, data);
+            this.external = null;
+        }
+    }
+}
+
+class Server extends EventEmitter {
+    constructor(options, callback) {
+        super();
+
+        if (!options) {
+            throw new TypeError('missing options');
+        }
+
+        if (options.port === undefined && !options.server && !options.noServer) {
+            throw new TypeError('invalid options');
+        }
+
+        var nativeOptions = WebSocketClient.PERMESSAGE_DEFLATE;
+
+        if (options.perMessageDeflate !== undefined) {
+            if (options.perMessageDeflate === false) {
+                nativeOptions = 0;
+            }
+        }
+
+        this.serverGroup = native.server.group.create(nativeOptions, options.maxPayload === undefined ? DEFAULT_PAYLOAD_LIMIT : options.maxPayload);
+
+        // can these be made private?
+        this._upgradeCallback = noop;
+        this._upgradeListener = null;
+        this._noDelay = options.noDelay === undefined ? true : options.noDelay;
+        this._lastUpgradeListener = true;
+        this._passedHttpServer = options.server;
+
+        if (!options.noServer) {
+            this.httpServer = options.server ? options.server : http.createServer((request, response) => {
+                // todo: default HTTP response
+                response.end();
+            });
+
+            if (options.path && (!options.path.length || options.path[0] !== '/')) {
+                options.path = '/' + options.path;
+            }
+
+            this.httpServer.on('upgrade', this._upgradeListener = ((request, socket, head) => {
+                if (!options.path || options.path == request.url.split('?')[0].split('#')[0]) {
+                    if (options.verifyClient) {
+                        const info = {
+                            origin: request.headers.origin,
+                            secure: request.connection.authorized !== undefined || request.connection.encrypted !== undefined,
+                            req: request
+                        };
+
+                        if (options.verifyClient.length === 2) {
+                            options.verifyClient(info, (result, code, name) => {
```

Comment:
__[jshint]__
* 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

<!-- Comment by [SideCI](https://sideci.com) -->